AsyncDisplayKit/ASPagerNode.m

```diff
@@ -84,6 +84,17 @@ - (ASCellNode *)collectionView:(ASCollectionView *)collectionView nodeForItemAtI
   return pageNode;
 }
 
+- (ASDataControllerCellNodeBlock)collectionView:(ASCollectionView *)collectionView nodeBlockAtIndexPath:(NSIndexPath *)indexPath {
+  ASDisplayNodeAssert(_pagerDataSource != nil, @"ASPagerNode must have a data source to load nodes to display");
+  if (![_pagerDataSource respondsToSelector:@selector(pagerNode:nodeBlockAtIndex:)]) {
```

Comment:
I actually don't believe this is premature since this value doesn't change until another dataSource is assigned. It would be nice to establish a pattern to abstraction for caching these calls